spec/lib/deeply_nested_params_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
+
+RSpec.describe DeeplyNestedParams do
+  let(:env) { Rack::MockRequest.env_for }
+  let(:app) { -> (_env) { [200, {}, ['success']] } }
+
+  subject { DeeplyNestedParams.new(app) }
+
+  it 'returns 200 response' do
+    status, _headers, _response = subject.call(env)
+    expect(status).to eq(200)
+  end
+
+  context 'if Rack::Utils.parse_query raises an RangeError' do
+    before do
+      allow(Rack::Utils).to receive(:parse_query).and_raise(RangeError)
```

Comment:
Should try to use known bad data passed to `#call` rather than relying on a stub, in case underlying implementation changes.

Reply:
Ah I remember now, Rack::Mock actually calls this method so if we attempt to pass in bad params then the `RangeErrror` exception happens too early.

https://github.com/rack/rack/blob/8be612ab949cf2eba7f4231ef17052a68315f911/lib/rack/mock.rb#L129-L130

Reply:
Ah, annoying. Okay let's go with it as is.